package proto_ff;

// calm 自定义的fieldoption都在该文件中
import "yd_fieldoptions.proto";
import "common_logic.proto";

import "Com.proto";

//服务器和服务器之间的协议号定义 5000-100000
enum ServerInternalCmd2
{
  //登录
  SERVER_INTERNAL2_BEGIN = 5000;
  WORLD_TO_LOGIC_GET_ROLE_LIST_REQ = 5001; //获取服务器角色列表
  LOGIC_TO_WORLD_GET_ROLE_LIST_RSP = 5002; //获取服务器角色列表
  WORLD_TO_LOGIC_CREATE_ROLE_INFO_REQ = 5003; //创建角色列表
  LOGIC_TO_WORLD_CREATE_ROLE_INFO_RSP = 5004; //创建角色列表
  WORLD_TO_PROXY_SERVER_BUS_ID_CHANGE_NOTIFY = 5005; //通知网关服务器BUSID修改
  WORLD_TO_OTHER_SERVER_NOTIFY_PLAYER_DISCONNECT = 5006; //通知Server玩家掉线
  WORLD_TO_LOGIC_LOGIN_REQ = 5007;
  LOGIC_TO_WORLD_LOGIN_RSP = 5008;
  WORLD_TO_SNS_LOGIN_REQ = 5009; //通知SnsServer玩家登陆
  SNS_TO_WORLD_LOGIN_RSP = 5010; //通知SnsServer玩家登陆
  GAME_TO_WORLD_REGISTER_MAP_REQ = 5011; //
  WORLD_TO_GAME_RESITER_MAP_RSP = 5012;
  WORLD_TO_LOGIC_LOGIN_FINISH_NOTIFY = 5013;
  LOGIC_TO_WORLD_ENTER_SCENE_REQ = 5014;
  WORLD_TO_LOGIC_ENTER_SCENE_RSP = 5015;
  WORLD_TO_GAME_ENTER_SCENE_REQ = 5016;
  GAME_TO_WORLD_ENTER_SCENE_RSP = 5017;

  LOGIC_TO_WORLD_LEAVE_SCENE_REQ = 5018;
  WORLD_TO_LOGIC_LEAVE_SCENE_RSP = 5019;
  WORLD_TO_GAME_LEAVE_SCENE_REQ = 5020;
  GAME_TO_WORLD_LEAVE_SCENE_RSP = 5021;

  LOGIC_TO_WORLD_GET_MAP_INFO_REQ = 5022;
  WORLD_TO_LOGIC_GET_MAP_INFO_RSP = 5023;
};